src/main/scala/com/crealytics/spark/excel/WorkbookReader.scala

```diff
@@ -56,26 +56,48 @@ object WorkbookReader {
       FileSystem.get(path.toUri, hadoopConfiguration).open(path)
     }
     parameters match {
-      case WithLocationMaxRowsInMemoryAndPassword(Seq(location), Seq(Some(maxRowsInMemory), passwordOption)) =>
-        new StreamingWorkbookReader(readFromHadoop(location), passwordOption, maxRowsInMemory.toInt)
-      case WithLocationMaxRowsInMemoryAndPassword(Seq(location), Seq(None, passwordOption)) =>
-        new DefaultWorkbookReader(readFromHadoop(location), passwordOption)
+      case WithLocationMaxRowsInMemoryAndPassword(
+            Seq(location),
+            Seq(Some(maxRowsInMemory), passwordOption, maxByteArraySizeOption)
+          ) =>
+        new StreamingWorkbookReader(
+          readFromHadoop(location),
+          passwordOption,
+          maxRowsInMemory.toInt,
+          maxByteArraySizeOption.map(_.toInt)
+        )
+      case WithLocationMaxRowsInMemoryAndPassword(Seq(location), Seq(None, passwordOption, maxByteArraySizeOption)) =>
+        new DefaultWorkbookReader(readFromHadoop(location), passwordOption, maxByteArraySizeOption.map(_.toInt))
     }
   }
 }
-class DefaultWorkbookReader(inputStreamProvider: => InputStream, workbookPassword: Option[String])
-    extends WorkbookReader {
+class DefaultWorkbookReader(
+  inputStreamProvider: => InputStream,
+  workbookPassword: Option[String],
+  maxByteArraySize: Option[Int]
+) extends WorkbookReader {
 
-  protected def openWorkbook(): Workbook =
+  protected def openWorkbook(): Workbook = {
+    maxByteArraySize.foreach { maxSize =>
+      IOUtils.setByteArrayMaxOverride(maxSize)
+    }
     workbookPassword
       .fold(WorkbookFactory.create(inputStreamProvider))(password =>
         WorkbookFactory.create(inputStreamProvider, password)
       )
+  }
 }
 
-class StreamingWorkbookReader(inputStreamProvider: => InputStream, workbookPassword: Option[String], maxRowsInMem: Int)
-    extends WorkbookReader {
+class StreamingWorkbookReader(
+  inputStreamProvider: => InputStream,
+  workbookPassword: Option[String],
+  maxRowsInMem: Int,
+  maxByteArraySize: Option[Int]
+) extends WorkbookReader {
   override protected def openWorkbook(): Workbook = {
+    maxByteArraySize.foreach { maxSize =>
+      IOUtils.setByteArrayMaxOverride(maxSize)
```

Comment:
Does this parameter work in `StreamingReader` as well?

Reply:
yes - excel-streaming-reader uses some POI code under the hood